netcore/System.Private.CoreLib/src/System.IO/Stream.cs

```diff
@@ -2,12 +2,16 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Runtime.CompilerServices;
+
 namespace System.IO
 {
 	partial class Stream
 	{
-		bool HasOverriddenBeginEndRead () => true;
+		[MethodImpl (MethodImplOptions.InternalCall)]
+		extern bool HasOverriddenBeginEndRead ();
```

Comment:
Wouldn't be better and maybe even simpler to make this intrinsic?

Reply:
Couldn't this be done entirely from managed?  Could get the `MethodInfo` for `BeginRead` from the base class and compare to the `MethodInfo` from `this.GetType()` ?

Reply:
I think the trick is where do you store the result to make it worth the optimization

Reply:
I guess it makes sense since we can emit offsets and slots as constants but I am not sure I know how to access vtable yet, will try.

Reply:
Converted into an intrinsic [WIP]

Reply:
(wouldn't say it's easier 😄)

Reply:
I would prefer an icall or doing it in managed. This has nothing to do with the JIT or anything low level.

Reply:
Most of the netcore intrinsics are there only for performance reasons not for correctness

Reply:
Anyway it was a good practice, this is the first time I emit basicblocks and conditions 🙂 I've just finished it, it passes my tests but I suspect the condition part can be simplified (but I don't know how). Here is the test I used: https://gist.github.com/EgorBo/48d0c78a7195dfb023c35a6c88362d10

Reply:
I gave on intrinsic and rolled back to the icall, intrinsic impl was too big and not aot friendly (couldn't hard-code base_klass address). I think the icall should be fine, the idea was to avoid allocations rather than nanoseconds and it's also an icall in CoreCLR. But still it was a good experience for me 🙂